includes/wizards/newspack/class-custom-events-section.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Custom Events Section Object.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack\Wizards\Newspack;
+
+/**
+ * WordPress dependencies
+ */
+use WP_Error, WP_REST_Server;
+
+/**
+ * Internal dependencies
+ */
+use Newspack\Wizards\Wizard_Section;
+
+/**
+ * Custom Events Section Object.
+ *
+ * @package Newspack\Wizards\Newspack
+ */
+class Custom_Events_Section extends Wizard_Section {
+
+	/**
+	 * Register Wizard Section specific endpoints.
+	 *
+	 * @return void 
+	 */
+	public function register_rest_routes() {
+		register_rest_route(
+			NEWSPACK_API_NAMESPACE_V2,
+			'/wizard/analytics/ga4-credentials',
+			[
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => [ $this, 'api_set_ga4_credentials' ],
+				'permission_callback' => [ $this, 'api_permissions_check' ],
+				'args'                => [
+					'measurement_id'              => [
+						'sanitize_callback' => 'sanitize_text_field',
+						'validate_callback' => [ $this, 'validate_measurement_id' ],
+					],
+					'measurement_protocol_secret' => [
+						'sanitize_callback' => 'sanitize_text_field',
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Updates the GA4 crendetials
+	 *
+	 * @param WP_REST_Request $request The REST request.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function api_set_ga4_credentials( $request ) {
+		$measurement_id              = $request->get_param( 'measurement_id' );
+		$measurement_protocol_secret = $request->get_param( 'measurement_protocol_secret' );
+
+		if ( ! $measurement_protocol_secret ) {
+			return new WP_Error(
+				'newspack_analytics_wizard_invalid_params',
+				esc_html__( 'Invalid Measurement Protocol API Secret.', 'newspack' ),
+				[ 'status' => 400 ]
+			);
+		}
+
+		update_option( 'ga4_measurement_id', sanitize_text_field( $measurement_id ) );
+		update_option( 'ga4_measurement_protocol_secret', sanitize_text_field( $measurement_protocol_secret ) );
+
+		return rest_ensure_response( $this->get_data() );
```

Comment:
I'm not familiar with how option caching works, but is it necessary to trigger two calls to `get_option` via `get_data` when we are already have the values to return in the response stored in variables above?

Reply:
While this is not strictly related to cache - options are auto-loaded (Unless specified in the 3rd arg in `*_option` functions) which means they're always in memory. I originally left the `$this->get_data()` call for these reasons + it also ensures the integrity of the data and only returning what is stored in the DB. 

Have removed and reused existing variables within the method.